sync_jira_actions/sync_to_jira.py

```diff
@@ -40,36 +40,44 @@ def applicationlinks(self):
 
 def main():
     if 'GITHUB_REPOSITORY' not in os.environ:
-        print('Not running in GitHub action context, nothing to do')
+        print('❌ Not running in GitHub action context, nothing to do')
         return
 
     if not os.environ['GITHUB_REPOSITORY'].startswith('espressif/'):
-        print('Not an Espressif repo, nothing to sync to JIRA')
+        print('❌ Not an Espressif repo, nothing to sync to JIRA')
         return
 
+    # For backward compatibility; handles situation if JIRA_PROJECT is not set in caller workflow either as input or ENV
+    if 'JIRA_PROJECT' not in os.environ:
+        print('❌ JIRA_PROJECT not set, fail!')
+        raise SystemExit(1)
```

Comment:
Just a thought: 

Let's agree how we do this company-wide. 

I would vote for `sys.exit(1)` in all the regular cases, when you just need to exit and have no intent to handle it further.

https://docs.python.org/3.12/library/sys.html#sys.exit

What is your opinion?

Reply:
@kumekay, the reason why I prefer `raise SystemExit ...` is that I don't need to import `sys`. Other than that, it is the same.  
Since more projects use `sys.exit`, I'll follow the crowd and change it in my projects too 👍🏻 .

BTW: I agree we should probably define it; this isn't the first time we've had this discussion :) Check out https://github.com/espressif/standards/discussions/2. I'll be glad for any thoughts and ideas.

Reply:
My 2c, I prefer code which uses `raise SystemExit` over `sys.exit` because such code is easier to reuse from another program. What happens often is, developers write CLI tools and use `sys.exit` there. Later if you want to reuse some code from that CLI tool in another tool, you end up having to monkeypatch sys.exit to prevent your top level tool from exiting... Using `raise SystemExit` everywhere seems to solve this issue.

Reply:
@igrr I believe, library code should never use either of these, but from a practical point of view there is no difference, i.e.:

```
try:
    sys.exit()
except SystemExit:
    print('handled')
```

will work the same.

In ESP-IDF codebase, there are 188 `sys.exit`s and 101 `SystemExit`.

Reply:
I prefer using custom SystemExit class. Benefits:
- could be caught on purpose
- uncaught errors will be printed automatically to stderr without traceback

```python
class Custom(SystemExit):
    pass

def fff():
    raise Custom('this is a error message')

fff()  # print to stderr in clients
```

Reply:
TIL, `SystemExit` will only be printed when `code` is unknown. https://github.com/python/cpython/blob/main/Python/pythonrun.c#L598

if we set

```python
class Custom(SystemExit):
    pass

def fff():
    err = Custom('this is a error message')
    err.code = 2
    raise err

fff()
```

The error message will not be printed anymore :D